$OpenBSD: patch-include_json_config_h,v 1.2 2016/06/03 13:25:26 dcoppa Exp $

Generalize setting of JSONCPP_OVERRIDE to all compilers
(upstream git commit ba6fa48d31091a151b10dd47986602889731a893)

--- include/json/config.h.orig	Fri Mar 25 21:09:15 2016
+++ include/json/config.h	Fri Jun  3 15:11:34 2016
@@ -77,12 +77,16 @@
 
 #endif // defined(_MSC_VER)
 
-#if defined(_MSC_VER) && _MSC_VER <= 1600 // MSVC <= 2010
-# define JSONCPP_OVERRIDE
-#else
+// In c++11 the override keyword allows you to explicity define that a function
+// is intended to override the base-class version.  This makes the code more
+// managable and fixes a set of common hard-to-find bugs.
+#if __cplusplus >= 201103L 
 # define JSONCPP_OVERRIDE override
-#endif // MSVC <= 2010
-
+#elif defined(_MSC_VER) && _MSC_VER > 1600
+# define JSONCPP_OVERRIDE override
+#else
+# define JSONCPP_OVERRIDE
+#endif
 
 #ifndef JSON_HAS_RVALUE_REFERENCES
 
